src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ComService.kt

```diff
@@ -96,6 +99,33 @@ class ComService(
       communityApiClient.getTeamsCodesForUser(body.staffIdentifier),
       probationSearchApiSortBy,
     )
-    return entityProbationSearchResult.transformToModelProbationResult()
+
+    val enrichedProbationSearchResult = entityProbationSearchResult.mapNotNull {
+      val licences =
+        licenceRepository.findAllByCrnAndStatusCodeIn(it.identifiers.crn, LicenceStatus.IN_FLIGHT_LICENCES)
+
+      // If an empty list has been returned, there are no relevant licences relating to search for the offender
+      if (licences.isEmpty()) {
+        null
+      } else {
+        val nonActiveLicenceStatuses = LicenceStatus.IN_FLIGHT_LICENCES - LicenceStatus.ACTIVE
+
+        val currentLicence =
+          if (licences.size > 1) licences.find { licence -> licence.statusCode in nonActiveLicenceStatuses } else licences.first()
+
+        val isOnProbation = currentLicence?.statusCode?.isOnProbation()
+
+        transformToModelEnrichedSearchResult(it, currentLicence, isOnProbation)
```

Comment:
would probably make this an extension method on `it`. 
(You don't need to pass  `isOnProbation` as a separate field as you can access this on licence and this is just transformational logic - copying from the source field on to the destination field)

Reply:
Done